src/game/AI/ScriptDevAI/scripts/eastern_kingdoms/shadowfang_keep/shadowfang_keep.cpp

```diff
@@ -86,7 +86,11 @@ struct npc_shadowfang_prisonerAI : public npc_escortAI
                 break;
             case 12:
                 if (m_uiNpcEntry == NPC_ASH)
-                    DoCastSpellIfCan(m_creature, SPELL_UNLOCK);
+                    {
+                        m_creature->GetMotionMaster()->PauseWaypoints(2000);
```

Comment:
Waypoint pause for a set amount of time should be handled in the waypoint path itself and thus, it should allow to also set the proper orientation there, making this whole change unneeded.

Reply:
technically setting the facing is also unneeded, if the waypoint terminated while the character was standing directly in front of the door, having come from a waypoint that is centered and perpendicular to the door, so I guess all that's really needed is to rework the waypoints.

I don't have any proper tools for that, tho :/